js-code-guidelines.md

```diff
@@ -73,39 +68,79 @@ For the majority of our JavaScript projects, our goals are to:
 
-Always run tests before pushing and PR'ing your code.
+- A Tech Lead directs the development of an entire ecosystem of modules (i.e js-ipfs, js-libp2p, js-ipld and js-multiformats), it has a complete understanding of the stack, the IPFS project, its goals and participates actively in the ROADMAP planning.
+- A Lead Maintainer is a contributor that has shown extraordinary ability to contribute to the project and willingness to make the project better by taking the responsibility of stewarding one of its modules forward.
+
+With this structure, we expect to achieve the following goals:
+
+- Recognize extraordinary contributions and empower contributors to take an even more important role in the project.
+- Reduce PR review time and Time To Release.
+- Increase the overall quality of the project.
+- Help users know who to reach out to for help.
+- Have a clear protocol to pass on the baton.
 
-### Captains and Maintainers
+The current Tech Leads are:
 
-For each JavaScript repository in the IPFS, libp2p, IPLD, or related GitHub orgs, there should be a single Captain specified in the Maintainers section of the README for that repository. The Captain is in charge of merging PRs, keeping issues up to date, and overall quality of a repository.
+- [David Dias](https://github.com/diasdavid/) for js-ipfs, js-libp2p js-multiformats ecosystems.
+- [Volker Mische](https://github.com/vmx) for the js-ipld ecosystem.
 
-Sometimes, a Captain may elect to have other maintainers that also have merge ability and commit access to the main repo. These maintainers can help out, but defer to the Captain as the person in charge of maintaining quality across the repo. It is important that this distinction is explicit; if there are long-standing PRs or issues, it is ultimately up to the Captain to gather information about the issue or PR. A Captain only makes a decision if he _needs_ to and _all_ methods of discussion are exhausted. Our community strives to trust the Captain as someone who ultimately has the most knowledge of a repo (even if they are also opinionated, and even if they have to spend effort to source that knowledge). This may change in the future, if we go with more non-hierarchical model.
+The current Lead Maintainers can be identified either by the `maintainer` field in the package.json of the module and/or the section `Maintainer` in the README.md of the module.
 
-Repositories which do not have Captains will show up in red on [Project Repos](https://project-repos.ipfs.io). Please open an issue in the repository asking for a captain to be nominated if none is provided.
+**Lead Maintainer responsibilities:**
 
-If you would like to become a maintainer (or lieutenant, or first mate, or whatever nautical metaphor you prefer), please open an issue about it! Chances are that we would love to have more help.
+- Respect and follow the [IPFS Code of Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md).
```

Comment:
Is this supposed to mean "enforce" rather than follow? The CoC applies to everyone and might be unnecessary to have here.

Reply:
It is a healthy reminder.